### ISSUE TITLE
Resource Monitor start_timestamp string format.

### ISSUE THREAD

**Issue by abrougher1-rbi** [Snowflake-Labs/terraform-provider-snowflake#1175]
**Provider Version**

0.37.1

**Terraform Version**

v1.1.9

**Describe the bug**

The resource_monitor start_timestamp never matches forcing TF to replace the resource on each run.

`  # module.account.snowflake_resource_monitor.ingestion_monitor[0] must be replaced
-/+ resource "snowflake_resource_monitor" "ingestion_monitor" {
      ~ id                         = "INGESTION_MONITOR" -> (known after apply)
        name                       = "INGESTION_MONITOR"
      ~ start_timestamp            = "2022-08-17T00:00:00Z" -> "2022-08-17 00:00" # forces replacement
      - suspend_immediate_triggers = [] -> null
      - suspend_triggers           = [] -> null
        # (5 unchanged attributes hidden)
    }`
    
The issue looks like it might be due to the format of the start_timestamp when read from Snowflake vs the timestamp required to set the value in snowflake.

ResourceMonitor Show is likely returning the format "2022-08-17T00:00:00Z" while we set the string to "2022-08-17 00:00" in the config.
https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/resources/resource_monitor.go#L189

**Expected behavior**

The snowflake_resource_monitor resource should be created once and not be replaced on every plan.

**Additional context**

This issue is similar to a previously fixed resource_monitor type bug.
https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/145

**Comment by culpgrant** [comment]
Hey @abrougher1-rbi did you ever come up with a workaround?

**Comment by funes79** [comment]
Hi, 
 it looks like that the provider is generating a command like this:
```
CREATE OR REPLACE RESOURCE MONITOR "MY_WH" FREQUENCY='MONTHLY' START_TIMESTAMP='2023-03-01T00:00:000Z' CREDIT_QUOTA=100 TRIGGERS ON 130 PERCENT DO SUSPEND ON 135 PERCENT DO SUSPEND_IMMEDIATE ON 80 PERCENT DO NOTIFY ON 100 PERCENT DO NOTIFY
```
But it throws a Snowflake error:
```
Invalid date/time format string '2023-03-01T00:00:000Z': Invalid format: "2023-03-01T00:00:000Z" is malformed at "T00:00:000Z"
```
Workaround is to create the resource monitor WITH the start_time, and after apply, remove the start_time from the tf code. :-/

**Comment by MartinGuindon** [comment]
I'm also experiencing this bug.

I can create a resource monitor successfully with the following code:
```
resource "snowflake_resource_monitor" "daily_quota_monitor" {
  name                       = "Daily Quota Monitor"
  warehouses                 = toset(keys(local.warehouses))
  credit_quota               = 5
  frequency                  = "DAILY"
  start_timestamp            = "2023-03-01 00:00"
  notify_triggers            = [50, 75]
  suspend_triggers           = [95]
  suspend_immediate_triggers = [100]
  depends_on                 = [snowflake_warehouse.warehouse]
}
```

But any subsequent runs of the TF project will report non-matching values:

```
  ~ resource "snowflake_resource_monitor" "daily_quota_monitor" {
        id                         = "Daily Quota Monitor"
        name                       = "Daily Quota Monitor"
      ~ start_timestamp            = "2023-03-01T00:00:00Z" -> "2023-03-01 00:00"
        # (7 unchanged attributes hidden)
    }
```
Which obviously fails.

Right now the workaround is to update the TF code to use the value `2023-03-01T00:00:00Z`. That value doesn't work on creation:

```
│ Error: error creating resource monitor Daily Quota Monitor err = 002105 (42601): SQL compilation error:
│ Invalid date/time format string '2023-03-01T00:00:00Z': Invalid format: "2023-03-01T00:00:00Z" is malformed at "T00:00:00Z"
│ 
│   with snowflake_resource_monitor.daily_quota_monitor,
│   on resource_monitors.tf line 14, in resource "snowflake_resource_monitor" "daily_quota_monitor":
│   14: resource "snowflake_resource_monitor" "daily_quota_monitor" {
│ 
╵
Operation failed: failed running terraform apply (exit 1)
```

**Comment by sfc-gh-jcieslak** [comment]
Hi all 👋 
I'm currently working on the resource monitor as part of [preparing GA objects for V1](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#preparing-essential-ga-objects-for-the-provider-v1). This is a tricky issue to solve on the provider as Snowflake accepts multiple timestamp formats. In the refactored resource monitor (that should be available in v0.96.0), the format of `RRRR-MM-DD hh:mm` should not produce any plans, and it will most likely be the recommended format. Everything will be documented; I'll let you know here when the v0.96.0 will be available so you can try the refreshed resource monitor.
